### ISSUE TITLE
Crash with 'string|callable' param type

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#256]
```php
<?php
/**
 * @param string|callable $param
 */
function f($param): void {}
f('is_array');
```
results in the following exception:
```
PHP Fatal error:  Uncaught UnexpectedValueException: There must be more than one unique type in /home/weirdan/.composer/vendor/vimeo/psalm/src/Psalm/Checker/TypeChecker.php:1302
Stack trace:
#0 /home/weirdan/.composer/vendor/vimeo/psalm/src/Psalm/Checker/Statements/Expression/CallChecker.php(2168): Psalm\Checker\TypeChecker::simplifyUnionType(Object(Psalm\Checker\ProjectChecker), Object(Psalm\Type\Union))
#1 /home/weirdan/.composer/vendor/vimeo/psalm/src/Psalm/Checker/Statements/Expression/CallChecker.php(1813): Psalm\Checker\Statements\Expression\CallChecker::checkFunctionArgumentType(Object(Psalm\Checker\StatementsChecker), Object(Psalm\Type\Union), Object(Psalm\Type\Union), 'f', 0, Object(Psalm\CodeLocation), Object(PhpParser\Node\Scalar\String_), Object(Psalm\Context))
#2 /home/weirdan/.composer/vendor/vimeo/psalm/src/Psalm/Checker/Statements/Expression/CallChecker.php(317): Psalm\Checker\Statements\Expression\CallChecker::checkFunctionArgumentsMatch(Object(Psalm\Checker\StatementsChecker), Array, 'f', Array, Object(Psalm\ in /home/weirdan/.composer/vendor/vimeo/psalm/src/Psalm/Checker/TypeChecker.php on line 1302
```
Modifying the param type to `string|object|callable` results in the following false positive:
```
ERROR: InvalidArgument - qq.php:9:3 - Argument 1 of f expects object, string provided
f('is_array');
```

**Comment by muglug** [comment]
I've fixed the crashing, but not the false-positive. Could you open a separate ticket?

**Comment by weirdan** [comment]
Sure. Thought they were closely related.

This change eliminated the false-positive I had in the real use case though:
```php
// namespace Zend\ServiceManager
// /**
//  * @param string $name
//  * @param string|callable|Factory\FactoryInterface $factory
//  */
// public function setFactory($name, $factory)

$container->setFactory($name, FactoryImpl::class);
```
